cmd/pkg/keymgmt/keymgmt.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright © 2021 Luke Hinds <lhinds@redhat.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package keymgmt
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+)
+
+func GeneratePrivateKey(algorithm string) (interface{}, interface{}, error) {
+	var err error = nil
+	var key interface{}
+	// Allow algorithm agility if different projects have certain FIPS like compliance requirements
+	switch algorithm {
+	case "1024":
+		key, err = rsa.GenerateKey(rand.Reader, 1024)
+	case "2048":
+		key, err = rsa.GenerateKey(rand.Reader, 2048)
+	case "3072":
+		key, err = rsa.GenerateKey(rand.Reader, 3072)
+	case "P224":
+		key, err = ecdsa.GenerateKey(elliptic.P224(), rand.Reader)
+	case "P256":
+		key, err = ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	case "P384":
+		key, err = ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
+	case "P521":
+		key, err = ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
```

Comment:
I'd also be in favor of adding ed25519 if we're going to add all of these :)

Reply:
That curve is not in the library (unless I am looking in the wrong place): 

https://golang.org/pkg/crypto/elliptic/

Reply:
ok, got you https://golang.org/pkg/crypto/ed25519/

do you mind if I do that in a follow up PR?

Reply:
Sure!

Reply:
https://golang.org/pkg/crypto/ed25519/ for reference - it's a separate package.